### PR TITLE
OMID-198 Replace static ports used for TSO server with random ports i…

### PR DESCRIPTION
**Pull request by chrajeshbabu**
…n the tests(Rajeshbabu)

### CODE REVIEW

**Comment by chrajeshbabu** [comment]
@stoty took some simple step of creating random number between 1050 to 20000 and use as a random port number in the tests. WDYT?

**Comment by stoty** [comment]
Usually this construct is used to generate a free port:

https://github.com/apache/hbase/blob/01876071b990063899e0839cdda9745783fe814c/hbase-http/src/test/java/org/apache/hadoop/hbase/http/HttpServerFunctionalTest.java#L278

**Comment by chrajeshbabu** [comment]
> Usually this construct is used to generate a free port:
> 
> https://github.com/apache/hbase/blob/01876071b990063899e0839cdda9745783fe814c/hbase-http/src/test/java/org/apache/hadoop/hbase/http/HttpServerFunctionalTest.java#L278

yes in HBase or Hadoop port 0 will be used to get free port. Let me check the same way.

**Comment by chrajeshbabu** [comment]
@stoty handled the review comment and fixed the failing test cases. Please take a look. Thanks

File: hbase-client/src/test/java/org/apache/omid/transaction/OmidTestBase.java [hunk 1]
```diff
@@ -99,12 +102,12 @@ public void beforeGroups(ITestContext context) throws Exception {
...
         tso.startAsync();
         tso.awaitRunning();
-        TestUtils.waitForSocketListening("localhost", 1234, 100);
+        TestUtils.waitForSocketListening("localhost", port, 100);
         LOG.info("Finished loading TSO");
         context.setAttribute("tso", tso);
 
         OmidClientConfiguration clientConf = new OmidClientConfiguration();
-        clientConf.setConnectionString("localhost:1234");
+        clientConf.setConnectionString("localhost:"+port);
```
Comment (stoty) [inline hbase-client/src/test/java/org/apache/omid/transaction/OmidTestBase.java:110]: please add spaces around "+" here, and every such occurence

File: hbase-client/src/test/java/org/apache/omid/transaction/TestFilters.java [hunk 2]
```diff
@@ -71,9 +71,8 @@ public void testGetWithValueFilter(ITestContext context) throws Exception {
     private void testGet(ITestContext context, Filter f) throws Exception {
 
         CommitTable.Client commitTableClient = spy(getCommitTable(context).getClient());
-
         HBaseOmidClientConfiguration hbaseOmidClientConf = new HBaseOmidClientConfiguration();
-        hbaseOmidClientConf.setConnectionString("localhost:1234");
+        hbaseOmidClientConf.setConnectionString("localhost:"+port+"");
```
Comment (stoty) [inline hbase-client/src/test/java/org/apache/omid/transaction/TestFilters.java:75]: The last +"" is not needed, please remove it everywhere in the patch.

File: tso-server/src/test/java/org/apache/omid/tso/TestTSOChannelHandlerNetty.java [hunk 3]
```diff
@@ -70,186 +72,197 @@
...
+            channelHandler.reconnect();
+            assertTrue(channelHandler.listeningChannel.isOpen());
+            assertEquals(channelHandler.channelGroup.size(), 1);
+
+            // Exercise closeable with re-connection trial
+            channelHandler.close();
+            assertFalse(channelHandler.listeningChannel.isOpen());
+            assertEquals(channelHandler.channelGroup.size(), 0);
+            try {
+                channelHandler.reconnect();
```
Comment (stoty) [inline tso-server/src/test/java/org/apache/omid/tso/TestTSOChannelHandlerNetty.java:126]: If this is not supposed to succed, then please an Assert.fail after.

**Review by stoty — CHANGES_REQUESTED** [review]
Looks mostly good, but please address the minor stylistic issues.

**Review by stoty — APPROVED** [review]
+1 LGTM

**Pull request closed without merging**